### PR TITLE
Patch/clarify webhook

### PR DESCRIPTION
**Pull request by robertjahn**

### CODE REVIEW

**Comment by robertjahn** [comment]
This will also address https://github.com/keptn/keptn.github.io/issues/1037

**Review by oleg-nenashev — COMMENTED** [review]
The addition to the main page looks nice! Just a style thing, maybe it makes sense to split long paragraphs to multiple lines like it was in the original content. It is useful for code reviews and for incremental pull requests.

I also wonder about removing the Jenkins webhook example. It seems to be a nice addition if this is still relevant. Should it be rather moved to Tutorials and referenced from there? Same for Slack that it rather on ArtifactHub now

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Requesting changes to indicate there is a discussion to be done @robertjahn

**Comment by agardnerIT** [comment]
More than happy to be overruled but I have a problem with the `interactive` and `non interactive` webhook terminology. The first time I heard those terms I thought **I** had to interact to do something.

Perhaps add some text like: "If you would like a third party too to **act**, you most likely want to subscribe to the `.triggered` event to trigger your external tool. In this case, the external tool is responsible for sending a `.started` and `.finished` event back to Keptn.
If you want to notify a tool that something has occurred (for example notify a messaging platform) , you'd usually subscribe to a `.finished` event. In this case, the integration is fire-and-forget so you don't need to send any other events."

**Comment by grabnerandi** [comment]
Chiming in on the discussion regarding interactive and non-interactive. How about "active" and "passive" integrations? Active means that the integration actively applies some changes to the environment, e.g: deploy, execute a test, ... -> passive on the other hand is just consuming the data without changing the impact of the sequence flow.
besides active & passive you could also use "Actor" & "Consumer"

Just my thoughts as a non-native speaker that also thinks that interactive and non-interactive sounds a bit weird

**Comment by robertjahn** [comment]
I agree the name should be easy to refer to and clear what is means.  Some options.  I am leaning towards the one with stars because they do describe better how they work, but they are kinda long in the name.

ONEWAY
a) Asynchronous
b) unidirectional
c) fire-and-forget *
d) silent
e) non-interactive
f) passive
g) broadcast

TWOWAY
a) Synchronous
b) request-response
c) Synchronous request-response *
d) bidirectional
e) interactive
f) active

here is a good references I found for pattern types: https://docs.oracle.com/cd/E23943_01/doc.1111/e17363/chapter05.htm#FPCON244 
https://www.enterpriseintegrationpatterns.com/patterns/conversation/BasicIntro.html

**Comment by miigwi** [comment]
I like the "active" / "passive" differentiation for the webhook integrations. Gives a better picture than "interactive" and "non-interactive".

**Comment by johannes-b** [comment]
I like the idea @grabnerandi was proposing: differentiation between *active* and *passive*. 
These two terms are short, precise, and bring it to the point.

**Comment by robertjahn** [comment]
ok seem to be consensus on active and passive.  I have adjusted accordingly in my last commit

**Comment by johannes-b** [comment]
@robertjahn please do not delete: content/docs/0.13.x/integrations/webhooks/jenkins/index.md 
We still need it since there is no Jenkins example on the ArtifactHub for Jenkins: 
image

File: content/docs/0.13.x/integrations/webhooks/_index.md [hunk 1]
```diff
@@ -5,74 +5,96 @@ weight: 2
...
-custom HTTP endpoints when running a delivery or remediation sequence that triggers a certain task. By using this
-integration, you can easily send the state of a task to a third-party tool or service. This allows you to integrate
-various tools such as testing services, existing CI/CD pipelines, and incident management services.
+Keptn has a built-in capability, via the [Keptn webhook service](https://github.com/keptn/keptn/tree/master/webhook-service), to call external HTTP endpoints as part of sequence task orchestration. 
 
+Webhooks allow you to easily integrate various third-party tools such as testing services, CI/CD pipelines, and incident management services. In addition to just invoking the third-party webhook receiver, webhook subscriptions can be configured to wait for the third-party webhook receiver to send back details about what it did. 
 
-## Create a Webhook integration
+Depending on the receiving tools configuration requirements, data can be either passed within the payload body or as query params in the URL along with any required headers and API tokens.
```
Comment (johannes-b) [inline content/docs/0.13.x/integrations/webhooks/_index.md:12]: Should be: "... receiving tool's configuration ..."

File: content/docs/0.13.x/integrations/webhooks/_index.md [hunk 2]
```diff
@@ -5,74 +5,96 @@ weight: 2
...
 
-Webhooks are created at a *Task* level and can be triggered by the following [event types](https://github.com/keptn/spec/blob/0.2.2/cloudevents.md#task-events):
+## Webhook subscriptions
+
+Webhook configurations, referred to as subscriptions, are created at a *Task* level and are configured to listen to one of the the following task [event types](https://github.com/keptn/spec/blob/0.2.2/cloudevents.md#task-events):
```
Comment (johannes-b) [inline content/docs/0.13.x/integrations/webhooks/_index.md:16]: Remove: ~~the~~ the following

File: content/docs/0.13.x/integrations/webhooks/_index.md [hunk 3]
```diff
@@ -5,74 +5,96 @@ weight: 2
...
+width="500px">}}
 
 *Subscription:*
 
 * *Task*: The task the webhook should be fired on (e.g., `test` or `deployment`)
-* *Task suffix*: The state of the task when the webhook should be fired; select one of: `triggered`, `started`, of `finished`
-* *Filter*: To restrict the webhook to certain stages and services you can specify those using filters.
-
-*Note:* Custom data in the event payload is available only for `triggered` task suffix.
+* *Task suffix*: The state of the task when the webhook should be fired; select one of: `started`, of `finished`
```
Comment (johannes-b) [inline content/docs/0.13.x/integrations/webhooks/_index.md:57]: `, of finished ` should be  `, or finished`

File: content/docs/0.13.x/integrations/webhooks/_index.md [hunk 4]
```diff
@@ -82,112 +104,194 @@ Based on the Go templating capabilities, you can:
...
-caption="Select event data to customize the request payload"
+caption="Select event data to add to the payload or URL"
 width="700px">}}
 
-</p>
-</details>
+**Step 4: Set "Send Finished Event" flag**
 
-## Include sensitive data
+The `Send finished` event flag is not applicable to a *non-interative* webhook. So you can leave it unselected as shown below.
```
Comment (johannes-b) [inline content/docs/0.13.x/integrations/webhooks/_index.md:118]: Please change `non-interactive` to `passive`

File: content/docs/0.13.x/integrations/webhooks/_index.md [hunk 5]
```diff
@@ -82,112 +104,194 @@ Based on the Go templating capabilities, you can:
...
+1. Follow the same steps above to add a new webhook subscription, but configure it to listen to a Task suffix of `triggered`. 
+1. Ensure that that the `By webhook receiver` option is selected as shown below. When this is set, the webhook service will expect the webhook receiver to send back the task `finished` event.
+
+    {{< popup_image
+    link="./assets/finished-event-receiver.png"
+    caption="Configure"
+    width="400px">}}
+1. Click the **Create subscription** button to save and enable the webhook for your integration.
+
+**Step 2: third-party webhook receiver**
```
Comment (johannes-b) [inline content/docs/0.13.x/integrations/webhooks/_index.md:173]: third-party > Third-party

File: content/docs/0.13.x/integrations/webhooks/_index.md [hunk 6]
```diff
@@ -82,112 +104,194 @@ Based on the Go templating capabilities, you can:
...
-spec:
-  webhooks:
-    - type: sh.keptn.event.deployment.started
-      envFrom:
-        - name: secret_api_token
-          secretRef:
-            name: api
-            key: api-token
+1. Follow the same steps above to add a new webhook subscription, but configure it to listen to a Task suffix of `triggered`. 
+1. Ensure that that the `Automatically` option is selected as shown below. When this is set, the webhook service will automatically send the task `finished` event for your subscription.
```
Comment (johannes-b) [inline content/docs/0.13.x/integrations/webhooks/_index.md:138]: Ensure ~~that~~ that

File: content/docs/0.13.x/integrations/webhooks/_index.md [hunk 7]
```diff
@@ -82,112 +104,194 @@ Based on the Go templating capabilities, you can:
...
 
-## Advanced Webhook configuration
+An example *Active* webhook is to trigger a system to open an incident management ticket and send back the newly created incident ticket number within the task `finished` event so that the sequence can continue.
 
-**Prerequisite:** This requires access to the [upstream Git repo](../../manage/git_upstream/) in order to modify the webhook configuration files.
+*Important Note:* 
 
-For more advanced configuration options, you can modify the raw request declared as [curl](https://curl.se/) command. Therefore, you need to access the `webhook.yaml` config file in the Git repo you set for an upstream. In this Git repo, you find the `webhook.yaml` file based on the filters you selected on the task subscription, e.g., if a filter is set for stage *production*, go to the *production* branch.
+* If the third-party webhook receiver does NOT send back the task `finished` event, the sequence will remain in a waiting states and will not continue until it times out or the it receives the task `finished` event.
```
Comment (johannes-b) [inline content/docs/0.13.x/integrations/webhooks/_index.md:155]: .. in a waiting states ... > in a waiting state

Comment (johannes-b) [inline content/docs/0.13.x/integrations/webhooks/_index.md:155]: ... or ~~the~~ it receives the task ...

File: content/docs/0.13.x/integrations/webhooks/_index.md [hunk 8]
```diff
@@ -82,112 +104,194 @@ Based on the Go templating capabilities, you can:
...
-        - "curl --request POST --data '{\"text\":\"Evaluation {{.data.evaluation.result}} with a score of {{.data.evaluation.score}} \"}'
-          https://hooks.slack.com/services/{{.env.secretKey}}"
-```
+1. Follow the same steps above to add a new webhook subscription, but configure it to listen to a Task suffix of `triggered`  
+1. The custom payload should include the following data in order for the receiving tool to properly respond with a `finished` event:
+  * `{{.id}}` - required in the `triggeredid` attribute of the `finished` event
+  * `{{.shkeptncontext}}` - required in the `shkeptncontext` attribute of the `finished` event
+  * Optional, but depends on your implementation: `{{.data.project}}`, `{{.data.service}}`, `{{.data.stage}}`
+1. Follow the same steps above to add a new webhook subscription, but configure it to listen to a Task suffix of `triggered`. 
+1. Ensure that that the `By webhook receiver` option is selected as shown below. When this is set, the webhook service will expect the webhook receiver to send back the task `finished` event.
```
Comment (johannes-b) [inline content/docs/0.13.x/integrations/webhooks/_index.md:165]: Ensure ~~that~~ that

File: content/docs/0.13.x/integrations/webhooks/_index.md [hunk 9]
```diff
@@ -82,112 +104,194 @@ Based on the Go templating capabilities, you can:
...
+    "specversion": "1.0",
+    "type": "sh.keptn.event.[REPLACE WITH YOUR TASK NAME].finished",
+    "shkeptncontext": "REPLACE WITH SHKEPTNCONTEXT FROM THE TRIGGERED EVENT",
+    "triggeredid": "REPLACE WITH ID FROM THE TRIGGERED EVENT"
+  }
+  ```
+
+The third-party webhook receiver can also send back custom information within the `finished` event. 
+
+This custom data will be automatically appended to every remaining `triggered` task event in the sequence.  The data much be passed in a attribute with the name of the task.
```
Comment (johannes-b) [inline content/docs/0.13.x/integrations/webhooks/_index.md:205]: .. passed in a**n** attribute ...

**Comment by robertjahn** [comment]
> @robertjahn please do not delete: content/docs/0.13.x/integrations/webhooks/jenkins/index.md We still need it since there is no Jenkins example on the ArtifactHub for Jenkins: image

@johannes-b  -- I put Jenkins back.  But I removed the slack page.  But to make jenkins not so lonely at the top of the page, I added an examples subpage.  
image

**Review by johannes-b — APPROVED** [review]
lgtm

**Review by oleg-nenashev — APPROVED** [review]
LGTM, thanks for the patches!

**Pull request merged**